src/deluge/storage/audio/audio_file_manager.cpp

```diff
@@ -322,17 +321,33 @@ Error AudioFileManager::getUnusedAudioRecordingFilePath(String* filePath, String
 		}
 	}
 
-	error = filePath->concatenate("/REC");
-	if (error != Error::NONE) {
-		return error;
-	}
-	error = filePath->concatenateInt(highestUsedAudioRecordingNumber[folderID], 5);
-	if (error != Error::NONE) {
-		return error;
+	// default to putting it in the normal spot if the song isn't named
+	if (songName->isEmpty()) {
+		error = filePath->concatenate("/REC");
+		if (error != Error::NONE) {
+			return error;
+		}
+		error = filePath->concatenateInt(highestUsedAudioRecordingNumber[folderID], 5);
+		if (error != Error::NONE) {
+			return error;
+		}
+		error = filePath->concatenate(".WAV");
+		if (error != Error::NONE) {
+			return error;
+		}
+		highestUsedAudioRecordingNumber[folderID]++;
 	}
-	error = filePath->concatenate(".WAV");
-	if (error != Error::NONE) {
-		return error;
+	else {
+		char namedPath[255]{0};
+		snprintf(namedPath, sizeof(namedPath), "%s/%s/%s.wav", filePath->get(), songName->get(), channelName);
```

Comment:
we should probably always use the indexed format below, otherwise the first recording will be named with a different format which feels undesirable.

Reply:
Oops this is a different PR - I agree but I'll make the change there